packages/data-service/src/data-service.ts

```diff
@@ -2113,7 +2128,7 @@ class DataServiceImpl extends WithLogContext implements DataService {
       result = await raceWithAbort(start(session), abortSignal);
     } catch (err) {
       if (isCancelError(err)) {
-        void abort();
+        await abort();
```

Comment:
No idea why we didn't do this before.

Reply:
because we didn't want to throw if abort failed. now that you changed it, should we catch `about` error and then continue to throw `err`, which is relevant in our case?

Reply:
We already wrote `abort` in a way such that it won’t throw, and if it does, that’s a genuine programming error then that we _should_ raise, right?

Reply:
If your cancellable thing is multiple queries, right now you _will_ get unhandled rejections if you abort it. For example.